### PR TITLE
Add return types to silence Symfony 6.3 deprecations

### PR DESCRIPTION
**Pull request by mbabker**
| Q | A
| --- | ---
| Branch? | 2.x
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | N/A
| License | MIT
| Doc | N/A

Symfony 6.3 added a number of return types to the framework. This PR should cover all the ones that popped up when beta testing an application.

Because all affected classes are non-final/internal, doc block annotations are used.

### CODE REVIEW

File: DependencyInjection/Compiler/AssetsVersionCompilerPass.php [hunk 1]
```diff
@@ -29,6 +29,11 @@
  */
 class AssetsVersionCompilerPass extends AbstractCompilerPass
 {
+    /**
+     * {@inheritdoc}
+     *
+     * @return void
+     */
     public function process(ContainerBuilder $container)
```
Comment (dbu) [inline DependencyInjection/Compiler/AssetsVersionCompilerPass.php]:
```suggestion
    public function process(ContainerBuilder $container): void
```
as its not expected that anyone extends these compilar passes, i would prefer to add actual return tye declarations instead of the phpdoc

File: LiipImagineBundle.php [hunk 2]
```diff
@@ -38,6 +38,8 @@ class LiipImagineBundle extends Bundle
 {
     /**
      * {@inheritdoc}
+     *
+     * @return void
      */
     public function build(ContainerBuilder $container)
```
Comment (dbu) [inline LiipImagineBundle.php]:
```suggestion
    public function build(ContainerBuilder $container): void
```

File: DependencyInjection/LiipImagineExtension.php [hunk 3]
```diff
@@ -59,6 +59,8 @@ public function getConfiguration(array $config, ContainerBuilder $container): ?C
     }
 
     /**
+     * @return void
+     *
      * @see \Symfony\Component\DependencyInjection\Extension.ExtensionInterface::load()
      */
     public function load(array $configs, ContainerBuilder $container)
```
Comment (dbu) [inline DependencyInjection/LiipImagineExtension.php]:
```suggestion
    public function load(array $configs, ContainerBuilder $container): void
```

File: DependencyInjection/Compiler/DriverCompilerPass.php [hunk 4]
```diff
@@ -23,6 +23,8 @@ class DriverCompilerPass extends AbstractCompilerPass
 {
     /**
      * {@inheritdoc}
```
Comment (dbu) [inline DependencyInjection/Compiler/DriverCompilerPass.php]: imho we should remove these empty phpdoc everywhere (could be a separate MR), as they do not add information.

**Review by dbu — CHANGES_REQUESTED** [review]
thanks. i would prefer to add actual return type declarations as the compiler passes are not extension points.

**Comment by mbabker** [comment]
> thanks. i would prefer to add actual return type declarations as the compiler passes are not extension points.

Updated

**Review by dbu — APPROVED** [review]
awesome, thanks a lot!

**Comment by dbu** [comment]
https://github.com/liip/LiipImagineBundle/releases/tag/2.11.0
